fix(flags): Split hypercache verification into separate tasks

haacked opened the pull request:
## Problem

We're not getting any metrics around the Celery task: `verify_and_fix_hypercaches_task`. I suspect it's because this tasks is not finishing in the allocated time budgeted.

## Changes

Split the combined `verify_and_fix_hypercaches_task` into two independent tasks to give each cache its own time budget and enable better metrics:

- `verify_and_fix_flags_cache_task` (runs at minute 40)
- `verify_and_fix_team_metadata_cache_task` (runs at minute 20)

This prevents timeouts when verifying both caches exceeded the 60-minute limit, and allows independent monitoring of each cache's verification.

## How did you test this code?

Unit Tests

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->
<!-- Removing this section does not mean the changelog bot won't pick it up, because *some people* like to not use the template, so we can't rely on it existing. -->

haacked committed:
Split hypercache verification into separate tasks

Split the combined `verify_and_fix_hypercaches_task` into two independent tasks to give each cache its own time budget and enable better metrics:

- `verify_and_fix_flags_cache_task` (runs at minute 40)
- `verify_and_fix_team_metadata_cache_task` (runs at minute 20)

This prevents timeouts when verifying both caches exceeded the 60-minute limit, and allows independent monitoring of each cache's verification.

haacked committed:
Add distributed lock to hypercache verification tasks

Use cache.add() to prevent concurrent executions of the same
verification task. If a previous run is still in progress, the
next scheduled run skips execution.

Also extends time limits from 1 hour to 4 hours since the lock
now prevents overlap instead of relying on strict time limits.

github-actions[bot] committed:
test(backend): update query snapshots

dmarticus reviewed (approved):
one comment about structlog

haacked committed:
Use structlog kwargs instead of f-strings for logging

structlog prefers passing variables as kwargs for structured logging
rather than embedding them in f-strings.

haacked committed:
Address Copilot review feedback

- Move Redis URL check before lock acquisition to avoid holding a lock
  when no work will be done
- Add explicit expires_seconds=4h to match task time limits, ensuring
  long-running tasks aren't silently dropped

haacked commented on posthog/tasks/test/test_hypercache_verification.py:
```diff
@@ -1,89 +1,103 @@
+        verify_and_fix_team_metadata_cache_task()
 
-        # Should not call verification at all
         mock_run_verification.assert_not_called()
```
I think the pattern and behavior is well understood. We don't need to test Django internals.
